Reviews/302 - RFP Portal/1 - Portal Development/README.md

```diff
@@ -0,0 +1,143 @@
+Project functionality meets/exceeds acceptance criteria and operates without error | PASS
+
+# Unit / Automated Testing
+
+After installing the requirements and adding the necessary configuration settings by following the given instructions on the project's README, the reviewer was able to run the tests under the `RFPPortal_Tests` directory. All positive, negative, and additional path tests were observed to run successfully.
+
+```bash
+cd RFPPortal_Tests
+dotnet test --filter DisplayName~Authorization_Tests
+```
+
+![img.png](assets/Authorization_Tests.png)
+
+```bash
+dotnet test --filter DisplayName~BidController_Tests
+```
+
+![img.png](assets/BidController_Tests.png)
+
+```bash
+dotnet test --filter DisplayName~RfpController_Tests
+```
+
+![img.png](assets/RfpController_Tests.png)
+
+Requirement | Finding
+------------ | -------------
+Unit Tests - At least one positive path test | PASS
+Unit Tests - At least one negative path test | PASS
+Unit Tests - Additional path tests | PASS
+
+# Documentation
+
+### Code Documentation
+
+Reviewer observed that all critical functions of the project had low-level code documentation which could be used for automated documentation generation. Reviewer praises the OP for the intensive code-level documentation.
+
+Requirement | Finding
+------------ | -------------
+Code Documented | PASS
+
+### Project Documentation
+
+Reviewer observed that the project's README had detailed general and usage documentation, exceeding the requirements.
+
+Requirement | Finding
+------------ | -------------
+Usage Documented | PASS
+
+## Overall Conclusion on Documentation
+
+Reviewer concludes that the project has extensive overall documentation, at both the high and the low level.
+
+# Open Source Practices
+
+## Licenses
+
+The Project is released under the MIT License
+
+Requirement | Finding
+------------ | -------------
+OSI-approved open source software license | PASS
+
+## Contribution Policies
+
+Pull requests and Issues are enabled. However, the project does not contain a CONTRIBUTING policy. It is highly suggested that the OP should add a proper contribution policy to provide guidelines for the potential contributors as it is crucial for the sustainability of an open-source project. However, this is not a reason to fail the whole submission, in the reviewer's opinion.
+
+Requirement | Finding
+------------ | -------------
+OSS contribution best practices | FAIL
+
+# Coding Standards
+
+## General Observations
+
+Code is generally well-structured and readable. The project as committed to GitHub and both the unit tests and the manual tests pass.
+
+# Final Conclusion
+
+The project provides the functionality described in the grant application and milestone acceptance criteria. It also meets or exceeds the standards required by the DEVxDAO. Thus, in the reviewer's opinion, this submission should pass.
+
+# Recommendation
+
+Recommendation | PASS
```

Comment:
Should this not be a PASS with Notes? Considering the FAIL on OSS Policies?

Reply:
Considered the same thing, then came to think that we give "pass with notes" for the issues that can be fixed/improved in a follow-up milestone, but consider them failure causes if it is a final milestone. In this case, decided to go with just PASS because the OSS policies never cause a failure. Does it make sense?

Reply:
Agreed on Pass with Notes!

Reply:
I think it could still be a PASS with Notes. It is something that we'd love to see improved (hence the note), but we still pass the milestone.

Reply:
Makes sense, but that means we need to abandon "no pass with notes at the final milestone" policy then (only for the cases where we suggest improvements)? I'm okay with that if we do.

Reply:
For the record: Based on the internal discussion in the Telegram group, we, as the CRDAO, have decided to abandon the "no pass with notes at the final milestone" policy (only for the cases where we suggest improvements rather than noting down the normally failing causes) because we now may be reviewing multiple milestones (including the final milestone) of a grant simultaneously which breaks the notion of a pass with notes, but we would like to still deliver notes even though they are not failure causes, in the hope that they may serve advancement of the open-source projects.

Reply:
The reviews have been updated according to the resolution above.